az storage blob batch-upload does not support --tags param

### Describe the bug

The `az storage blob batch-upload` command does not accept `--tags`


Since `az storage blob upload` supports `--tags` and it's looks that batch-upload is utilizing `upload_blob()`

### Related command

`az storage blob upload-batch --account-name x --destination x --destination-path x --source "x" --tier x--validate-content --auth-mode login --tags x=x`

### Errors

```
ERROR: The value for one of the HTTP headers is not in the correct format.
RequestId:xxx
Time:2024-02-08T23:20:44.1431571Z
ErrorCode:InvalidHeaderValue
headername:x-ms-if-tags
headervalue:x=x
```

### Issue script & Debug output

```log
2024-02-08T23:57:46.2205660Z DEBUG: cli.knack.cli: Command arguments: ['storage', 'blob', 'upload-batch', '--account-name', 'storageAccount', '--destination', 'destFolder', '--destination-path', 'destPath', '--source', 'D:\\a\\1\\s/folder/', '--tier', 'Cool', '--validate-content', '--auth-mode', 'login', '--tags', 'tag=tagValue', '--debug']
2024-02-08T23:57:46.2212193Z DEBUG: cli.knack.cli: __init__ debug log:
2024-02-08T23:57:46.2212912Z Cannot enable color.
2024-02-08T23:57:46.2213258Z DEBUG: cli.knack.cli: Event: Cli.PreExecute []
2024-02-08T23:57:46.2213723Z DEBUG: cli.knack.cli: Event: CommandParser.OnGlobalArgumentsCreate [<function CLILogging.on_global_arguments at 0x0000014D3723B6A0>, <function OutputProducer.on_global_arguments at 0x0000014D373C5DA0>, <function CLIQuery.on_global_arguments at 0x0000014D373EB920>]
2024-02-08T23:57:46.2214345Z DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableCreate []
2024-02-08T23:57:46.2215668Z DEBUG: cli.azure.cli.core: Modules found from index for 'storage': ['azure.cli.command_modules.storage']
2024-02-08T23:57:46.2216856Z DEBUG: cli.azure.cli.core: Loading command modules:
2024-02-08T23:57:46.2217551Z DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands
2024-02-08T23:57:46.4916699Z DEBUG: cli.azure.cli.core: storage                   0.270        59       273
2024-02-08T23:57:46.4917381Z DEBUG: cli.azure.cli.core: Total (1)                 0.270        59       273
2024-02-08T23:57:46.4936521Z DEBUG: cli.azure.cli.core: These extensions are not installed and will be skipped: ['azext_ai_examples', 'azext_next']
2024-02-08T23:57:46.4941457Z DEBUG: cli.azure.cli.core: Loading extensions:
2024-02-08T23:57:46.4942109Z DEBUG: cli.azure.cli.core: Name                  Load Time    Groups  Commands  Directory
2024-02-08T23:57:46.4942535Z DEBUG: cli.azure.cli.core: Total (0)                 0.000         0         0  
2024-02-08T23:57:46.4942870Z DEBUG: cli.azure.cli.core: Loaded 59 groups, 273 commands.
2024-02-08T23:57:46.4943226Z DEBUG: cli.azure.cli.core: Found a match in the command table.
2024-02-08T23:57:46.4944514Z DEBUG: cli.azure.cli.core: Raw command  : storage blob upload-batch
2024-02-08T23:57:46.4945026Z DEBUG: cli.azure.cli.core: Command table: storage blob upload
2024-02-08T23:57:46.4945368Z DEBUG: cli.azure.cli.core: remaining    :                     batch
2024-02-08T23:57:46.4945799Z DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreCommandTableTruncate [<function AzCliLogging.init_command_file_logging at 0x0000014D3A2922A0>]
2024-02-08T23:57:46.4946595Z DEBUG: cli.azure.cli.core.azlogging: metadata file logging enabled - writing logs to 'D:\a\_temp\.azclitask\commands\2024-02-08.23-57-46.storage_blob_upload-batch.3796.log'.
2024-02-08T23:57:46.4947933Z INFO: az_command_data_logger: command args: storage blob upload-batch --account-name {} --destination {} --destination-path {} --source {} --tier {} --validate-content --auth-mode {} --tags {} --debug
2024-02-08T23:57:46.4948479Z DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreArgumentLoad [<function register_global_subscription_argument.<locals>.add_subscription_parameter at 0x0000014D3A2CE5C0>]
2024-02-08T23:57:50.1594285Z DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
2024-02-08T23:57:50.1595705Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 659, in _get_attr
2024-02-08T23:57:50.1596775Z AttributeError: module 'azure.mgmt.storage.v2023_01_01.models' has no attribute 'ActiveDirectoryPropertiesAccountType'
2024-02-08T23:57:50.1597042Z 
2024-02-08T23:57:51.7076270Z DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
2024-02-08T23:57:51.7077701Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 659, in _get_attr
2024-02-08T23:57:51.7078326Z AttributeError: module 'azure.mgmt.storage.v2023_01_01.models' has no attribute 'ListKeyExpand'
2024-02-08T23:57:51.7078570Z 
2024-02-08T23:57:51.7177329Z DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
2024-02-08T23:57:51.7178714Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 659, in _get_attr
2024-02-08T23:57:51.7179329Z AttributeError: module 'azure.mgmt.storage.v2023_01_01.models' has no attribute 'CorsRuleAllowedMethodsItem'
2024-02-08T23:57:51.7179552Z 
2024-02-08T23:57:53.1136206Z DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostArgumentLoad []
2024-02-08T23:57:53.1136943Z DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostCommandTableCreate [<function register_ids_argument.<locals>.add_ids_arguments at 0x0000014D3A2F0680>, <function register_cache_arguments.<locals>.add_cache_arguments at 0x0000014D3A2F07C0>]
2024-02-08T23:57:53.1166908Z DEBUG: cli.knack.cli: Event: CommandInvoker.OnCommandTableLoaded []
2024-02-08T23:57:53.1167465Z DEBUG: cli.knack.cli: Event: CommandInvoker.OnPreParseArgs []
2024-02-08T23:57:53.1178142Z DEBUG: cli.knack.cli: Event: CommandInvoker.OnPostParseArgs [<function OutputProducer.handle_output_argument at 0x0000014D373C5E40>, <function CLIQuery.handle_query_parameter at 0x0000014D373EB9C0>, <function register_ids_argument.<locals>.parse_ids_arguments at 0x0000014D3A2F0720>]
2024-02-08T23:57:53.1378249Z DEBUG: cli.azure.cli.core.auth.persistence: build_persistence: location='D:\\a\\_temp\\.azclitask\\service_principal_entries.bin', encrypt=True
2024-02-08T23:57:53.1398097Z DEBUG: cli.azure.cli.core.auth.persistence: build_persistence: location='D:\\a\\_temp\\.azclitask\\msal_token_cache.bin', encrypt=True
2024-02-08T23:57:53.1408036Z DEBUG: cli.azure.cli.core.auth.binary_cache: load: D:\a\_temp\.azclitask\msal_http_cache.bin
2024-02-08T23:57:53.1414126Z DEBUG: urllib3.util.retry: Converted retries value: 1 -> Retry(total=1, connect=None, read=None, redirect=None, status=None)
2024-02-08T23:57:53.1426025Z DEBUG: msal.authority: openid_config = {.....}
2024-02-08T23:57:53.1429502Z DEBUG: msal.application: Broker enabled? False
2024-02-08T23:57:53.1429819Z DEBUG: msal.application: Region to be used: None
2024-02-08T23:57:53.1575958Z DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
2024-02-08T23:57:53.1576773Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 655, in _get_attr
2024-02-08T23:57:53.1577641Z   File "importlib\__init__.py", line 126, in import_module
2024-02-08T23:57:53.1578062Z   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
2024-02-08T23:57:53.1578360Z   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
2024-02-08T23:57:53.1578650Z   File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
2024-02-08T23:57:53.1578946Z   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
2024-02-08T23:57:53.1581228Z   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
2024-02-08T23:57:53.1581545Z   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
2024-02-08T23:57:53.1581840Z   File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
2024-02-08T23:57:53.1582333Z ModuleNotFoundError: No module named 'azure.multiapi.storagev2.blob.v2022_11_02.blob'
2024-02-08T23:57:53.1582731Z 
2024-02-08T23:57:53.1583041Z DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
2024-02-08T23:57:53.1583587Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 655, in _get_attr
2024-02-08T23:57:53.1584125Z   File "importlib\__init__.py", line 126, in import_module
2024-02-08T23:57:53.1585549Z   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
2024-02-08T23:57:53.1586148Z   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
2024-02-08T23:57:53.1586487Z   File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
2024-02-08T23:57:53.1589001Z ModuleNotFoundError: No module named 'azure.multiapi.storagev2.blob.v2022_11_02.file'
2024-02-08T23:57:53.1589219Z 
2024-02-08T23:57:53.1589756Z DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
2024-02-08T23:57:53.1590485Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 655, in _get_attr
2024-02-08T23:57:53.1590834Z   File "importlib\__init__.py", line 126, in import_module
2024-02-08T23:57:53.1591092Z   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
2024-02-08T23:57:53.1591380Z   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
2024-02-08T23:57:53.1591673Z   File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
2024-02-08T23:57:53.1591971Z   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
2024-02-08T23:57:53.1595425Z   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
2024-02-08T23:57:53.1595748Z   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
2024-02-08T23:57:53.1596048Z   File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
2024-02-08T23:57:53.1596538Z ModuleNotFoundError: No module named 'azure.multiapi.storagev2.blob.v2022_11_02.blob'
2024-02-08T23:57:53.1596759Z 
2024-02-08T23:57:53.1597066Z DEBUG: cli.azure.cli.core.profiles._shared: Traceback (most recent call last):
2024-02-08T23:57:53.1597618Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/profiles/_shared.py", line 655, in _get_attr
2024-02-08T23:57:53.1597966Z   File "importlib\__init__.py", line 126, in import_module
2024-02-08T23:57:53.1598246Z   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
2024-02-08T23:57:53.1598522Z   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
2024-02-08T23:57:53.1598823Z   File "<frozen importlib._bootstrap>", line 1126, in _find_and_load_unlocked
2024-02-08T23:57:53.1599122Z   File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
2024-02-08T23:57:53.1599417Z   File "<frozen importlib._bootstrap>", line 1204, in _gcd_import
2024-02-08T23:57:53.1599679Z   File "<frozen importlib._bootstrap>", line 1176, in _find_and_load
2024-02-08T23:57:53.1599975Z   File "<frozen importlib._bootstrap>", line 1140, in _find_and_load_unlocked
2024-02-08T23:57:53.1600438Z ModuleNotFoundError: No module named 'azure.multiapi.storagev2.blob.v2022_11_02.file'
2024-02-08T23:57:53.1600628Z 
2024-02-08T23:57:53.2469907Z DEBUG: cli.azure.cli.core.auth.credential_adaptor: CredentialAdaptor.get_token: scopes=('https://storage.azure.com/.default',), kwargs={}
2024-02-08T23:57:53.2470881Z DEBUG: cli.azure.cli.core.auth.msal_authentication: ServicePrincipalCredential.get_token: scopes=('https://storage.azure.com/.default',), kwargs={}
2024-02-08T23:57:53.2479046Z DEBUG: msal.telemetry: Generate or reuse correlation_id: 7b31ae69-e15a-43d2-9b53-0f54ed6e18f3
2024-02-08T23:57:53.2522001Z DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): login.microsoftonline.com:443
2024-02-08T23:57:53.8168880Z DEBUG: urllib3.connectionpool: https://login.microsoftonline.com:443 "POST /tenantId/oauth2/v2.0/token HTTP/1.1" 200 1316
2024-02-08T23:57:53.8178589Z DEBUG: msal.token_cache: event={
......removed auth things......
2024-02-08T23:57:53.8186282Z }
2024-02-08T23:57:53.8243288Z DEBUG: urllib3.connectionpool: Starting new HTTPS connection (1): storageAccount.blob.core.windows.net:443
2024-02-08T23:57:54.2289108Z DEBUG: urllib3.connectionpool: https://storageAccount.blob.core.windows.net:443 "PUT /destFolder/destPath/folder/foo/file.file HTTP/1.1" 400 334
2024-02-08T23:57:54.2397520Z DEBUG: cli.azure.cli.core.azclierror: Traceback (most recent call last):
2024-02-08T23:57:54.2398267Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\knack/cli.py", line 233, in invoke
2024-02-08T23:57:54.2398752Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 663, in execute
2024-02-08T23:57:54.2399216Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 726, in _run_jobs_serially
2024-02-08T23:57:54.2402615Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 718, in _run_job
2024-02-08T23:57:54.2403104Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/__init__.py", line 430, in new_handler
2024-02-08T23:57:54.2406306Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/__init__.py", line 430, in new_handler
2024-02-08T23:57:54.2407494Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/_exception_handler.py", line 17, in file_related_exception_handler
2024-02-08T23:57:54.2408514Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 697, in _run_job
2024-02-08T23:57:54.2409045Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/__init__.py", line 333, in __call__
2024-02-08T23:57:54.2409964Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/core/commands/command_operation.py", line 121, in handler
2024-02-08T23:57:54.2412103Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/operations/blob.py", line 523, in storage_blob_upload_batch
2024-02-08T23:57:54.2412676Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/util.py", line 328, in wrapper
2024-02-08T23:57:54.2413333Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/operations/blob.py", line 505, in _upload_blob
2024-02-08T23:57:54.2414848Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/cli/command_modules/storage/operations/blob.py", line 639, in upload_blob
2024-02-08T23:57:54.2415517Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/core/tracing/decorator.py", line 76, in wrapper_use_tracer
2024-02-08T23:57:54.2416051Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/multiapi/storagev2/blob/v2022_11_02/_blob_client.py", line 746, in upload_blob
2024-02-08T23:57:54.2416901Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/multiapi/storagev2/blob/v2022_11_02/_upload_helpers.py", line 196, in upload_block_blob
2024-02-08T23:57:54.2417519Z   File "D:\a\_work\1\s\build_scripts\windows\artifacts\cli\Lib\site-packages\azure/multiapi/storagev2/blob/v2022_11_02/_shared/response_handlers.py", line 189, in process_storage_error
2024-02-08T23:57:54.2417980Z   File "<string>", line 1, in <module>
2024-02-08T23:57:54.2418357Z azure.core.exceptions.HttpResponseError: The value for one of the HTTP headers is not in the correct format.
2024-02-08T23:57:54.2419249Z RequestId:d5485e1c-401e-0004-70ea-5af7b3000000
2024-02-08T23:57:54.2419527Z Time:2024-02-08T23:57:54.2138689Z
2024-02-08T23:57:54.2419687Z ErrorCode:InvalidHeaderValue
2024-02-08T23:57:54.2419912Z headername:x-ms-if-tags
2024-02-08T23:57:54.2420080Z headervalue:tag=tagValue
2024-02-08T23:57:54.2420844Z Content: <?xml version="1.0" encoding="utf-8"?><Error><Code>InvalidHeaderValue</Code><Message>The value for one of the HTTP headers is not in the correct format.
2024-02-08T23:57:54.2421509Z RequestId:d5485e1c-401e-0004-70ea-5af7b3000000
2024-02-08T23:57:54.2422115Z Time:2024-02-08T23:57:54.2138689Z</Message><HeaderName>x-ms-if-tags</HeaderName><HeaderValue>tag=tagValue</HeaderValue></Error>
2024-02-08T23:57:54.2422275Z 
2024-02-08T23:57:54.2422659Z ERROR: cli.azure.cli.core.azclierror: The value for one of the HTTP headers is not in the correct format.
2024-02-08T23:57:54.2422984Z RequestId:d5485e1c-401e-0004-70ea-5af7b3000000
2024-02-08T23:57:54.2423267Z Time:2024-02-08T23:57:54.2138689Z
2024-02-08T23:57:54.2423461Z ErrorCode:InvalidHeaderValue
2024-02-08T23:57:54.2423736Z headername:x-ms-if-tags
2024-02-08T23:57:54.2423921Z headervalue:tag=tagValue
2024-02-08T23:57:54.2424320Z ERROR: az_command_data_logger: The value for one of the HTTP headers is not in the correct format.
2024-02-08T23:57:54.2425256Z RequestId:d5485e1c-401e-0004-70ea-5af7b3000000
2024-02-08T23:57:54.2426366Z Time:2024-02-08T23:57:54.2138689Z
2024-02-08T23:57:54.2427227Z ErrorCode:InvalidHeaderValue
2024-02-08T23:57:54.2430141Z headername:x-ms-if-tags
2024-02-08T23:57:54.2430428Z headervalue:tag=tagValue
2024-02-08T23:57:54.2430847Z DEBUG: cli.knack.cli: Event: Cli.PostExecute [<function AzCliLogging.deinit_cmd_metadata_logging at 0x0000014D3A292520>]
2024-02-08T23:57:54.2431202Z INFO: az_command_data_logger: exit code: 1
2024-02-08T23:57:54.2431510Z INFO: cli.__main__: Command ran in 8.531 seconds (init: 0.507, invoke: 8.023)
2024-02-08T23:57:54.2966625Z INFO: telemetry.main: Begin splitting cli events and extra events, total events: 1
2024-02-08T23:57:54.2967609Z INFO: telemetry.client: Accumulated 0 events. Flush the clients.
2024-02-08T23:57:54.2968012Z INFO: telemetry.main: Finish splitting cli events and extra events, cli events: 1
2024-02-08T23:57:54.2973725Z INFO: telemetry.save: Save telemetry record of length 3736 in cache
2024-02-08T23:57:54.2975776Z INFO: telemetry.main: Begin creating telemetry upload process.
2024-02-08T23:57:54.2980535Z INFO: telemetry.process: Creating upload process: "C:\Program Files\Microsoft SDKs\Azure\CLI2\python.exe C:\Program Files\Microsoft SDKs\Azure\CLI2\Lib\site-packages\azure\cli\telemetry\__init__.pyc D:\a\_temp\.azclitask"
2024-02-08T23:57:54.3023814Z INFO: telemetry.process: Return from creating process
2024-02-08T23:57:54.3099179Z INFO: telemetry.main: Finish creating telemetry upload process.
2024-02-08T23:57:54.7673552Z ##[error]Script failed with exit code: 1
```

### Expected behavior

`blob batch-upload` should support --tags as `upload`

### Environment Summary

azure-cli                         2.56.0 *

core                              2.56.0 *
telemetry                          1.1.0

Extensions:
azure-devops                      0.26.0

Dependencies:
msal                            1.24.0b2
azure-mgmt-resource             23.1.0b2

Python location 'C:\Program Files\Microsoft SDKs\Azure\CLI2\python.exe'
Extensions directory 'C:\Program Files\Common Files\AzureCliExtensionDirectory'

Python (Windows) 3.11.5 (tags/v3.11.5:cce6ba9, Aug 24 2023, 14:38:34) [MSC v.1936 64 bit (AMD64)]

### Additional context

_No response_

--- a/src/azure-cli/azure/cli/command_modules/storage/_params.py
+++ b/src/azure-cli/azure/cli/command_modules/storage/_params.py
@@ -1066,7 +1066,7 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
         c.argument('length', type=int, help='Number of bytes to read from the stream. This is optional, but should be '
                                             'supplied for optimal performance. Cooperate with --data.', is_preview=True,
                    min_api='2019-02-02')
-        c.argument('overwrite', arg_type=get_three_state_flag(), arg_group="Additional Flags", is_preview=True,
+        c.argument('overwrite', arg_type=get_three_state_flag(), arg_group="Additional Flags",
                    help='Whether the blob to be uploaded should overwrite the current data. If True, blob upload '
                         'operation will overwrite the existing data. If set to False, the operation will fail with '
                         'ResourceExistsError. The exception to the above is with Append blob types: if set to False and the '
@@ -1103,20 +1103,24 @@ def load_arguments(self, _):  # pylint: disable=too-many-locals, too-many-statem
 
         c.argument('source', options_list=('--source', '-s'))
         c.argument('destination', options_list=('--destination', '-d'))
-        c.argument('max_connections', type=int,
+        c.argument('max_connections', type=int, arg_group="Additional Flags",
                    help='Maximum number of parallel connections to use when the blob size exceeds 64MB.')
         c.argument('maxsize_condition', arg_group='Content Control')
         c.argument('validate_content', action='store_true', min_api='2016-05-31', arg_group='Content Control')
-        c.argument('blob_type', options_list=('--type', '-t'), arg_type=get_enum_type(get_blob_types()))
-        c.extra('no_progress', progress_type, validator=add_progress_callback)
-        c.extra('tier', tier_type, is_preview=True, validator=blob_tier_validator_track2)
-        c.extra('overwrite', arg_type=get_three_state_flag(), is_preview=True,
+        c.argument('blob_type', options_list=('--type', '-t'), arg_type=get_enum_type(get_blob_types()),
+                   arg_group="Additional Flags")
+        c.extra('no_progress', progress_type, validator=add_progress_callback, arg_group="Additional Flags")
+        c.extra('tier', tier_type, is_preview=True, validator=blob_tier_validator_track2, arg_group="Additional Flags")
+        c.extra('overwrite', arg_type=get_three_state_flag(), arg_group="Additional Flags",
                 help='Whether the blob to be uploaded should overwrite the current data. If True, blob upload '
                      'operation will overwrite the existing data. If set to False, the operation will fail with '
                      'ResourceExistsError. The exception to the above is with Append blob types: if set to False and the '
                      'data already exists, an error will not be raised and the data will be appended to the existing '
                      'blob. If set overwrite=True, then the existing append blob will be deleted, and a new one created. '
                      'Defaults to False.')
+        c.extra('tags', arg_type=tags_type, arg_group="Additional Flags")
+        c.argument('metadata', arg_group="Additional Flags")
+        c.argument('timeout', arg_group="Additional Flags")
 
     with self.argument_context('storage blob download', resource_type=ResourceType.DATA_STORAGE_BLOB) as c:
         c.register_blob_arguments_track2()

--- a/src/azure-cli/azure/cli/command_modules/storage/_validators.py
+++ b/src/azure-cli/azure/cli/command_modules/storage/_validators.py
@@ -8,7 +8,7 @@ import os
 import argparse
 from ipaddress import ip_network
 
-from azure.cli.core.commands.validators import validate_key_value_pairs
+from azure.cli.core.commands.validators import validate_key_value_pairs, validate_tags
 from azure.cli.core.profiles import ResourceType, get_sdk
 from azure.cli.core.util import get_file_json, shell_safe_json_parse
 from azure.cli.core.azclierror import UnrecognizedArgumentError
@@ -1252,6 +1252,7 @@ def process_blob_upload_batch_parameters(cmd, namespace):
     get_content_setting_validator(t_blob_content_settings, update=False)(cmd, namespace)
     add_progress_callback(cmd, namespace)
     blob_tier_validator_track2(cmd, namespace)
+    validate_tags(namespace)
 
 
 def process_blob_delete_batch_parameters(cmd, namespace):

--- a/src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_batch_operations.py
+++ b/src/azure-cli/azure/cli/command_modules/storage/tests/latest/test_storage_batch_operations.py
@@ -87,7 +87,7 @@ class StorageBatchOperationScenarios(StorageScenarioMixin, LiveScenarioTest):
                                   local_folder)
 
     @ResourceGroupPreparer()
-    @StorageAccountPreparer()
+    @StorageAccountPreparer(kind='StorageV2')
     @StorageTestFilesPreparer()
     def test_storage_blob_batch_upload_scenarios(self, test_dir, storage_account_info):
         # upload files without pattern
@@ -102,6 +102,14 @@ class StorageBatchOperationScenarios(StorageScenarioMixin, LiveScenarioTest):
         self.storage_cmd('storage blob list -c {}', storage_account_info, container).assert_with_checks(
             JMESPathCheck('length(@)', 41))
 
+        # test upload-batch with tags
+        self.storage_cmd('storage blob upload-batch -s "{}" -d {} --overwrite --tags key1=val1 key2=val2',
+                         storage_account_info, test_dir, container)
+        result = self.storage_cmd('storage blob list -c {} --prefix apple --include t',
+                                  storage_account_info, container).get_output_in_json()
+        self.assertEqual(len(result), 10)
+        for blob in result:
+            self.assertEqual(blob["tags"], {"key1": "val1", "key2": "val2"})
 
         # upload files with pattern apple/*
         container = self.create_container(storage_account_info)
@@ -153,7 +161,7 @@ class StorageBatchOperationScenarios(StorageScenarioMixin, LiveScenarioTest):
                                   container, test_dir, current).get_output_in_json()
         self.assertEqual(len(result), 41)
 
-        #check result url
+        # check result url
         container = self.create_container(storage_account_info)
         result = self.storage_cmd('storage blob upload-batch -s "{}" -d {}', storage_account_info,
                          test_dir, container).get_output_in_json()

